contrib/container-chroot-agent/entrypoint-agent.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+
+set -x
+
+# mount FSs from the host VM to /chroot folder
+for i in proc sys run dev; do mkdir -p /chroot/$i; done
+mount --types proc /rootproc /chroot/proc
+mount --rbind /rootsys /chroot/sys
+mount --make-rslave /chroot/sys
+mount --rbind /rootdev /chroot/dev
+mount --make-rslave /chroot/dev
+mount --bind /rootrun /chroot/run
+mount --make-slave /chroot/run
+for i in bin lib lib64 sbin; do cp -a /rootfs/$i /chroot/; done # for the POC it's hardcoded for Amazon Linux for now
+
+# install cfengine in /cfengine in container and then bind mount to /chroot/var/cfengine so that contents are only visible/present in container/chroot and not host
+# hack to install cfengine-nova, logged ticket ENT-11403 to make quick-install script used later for installation
+# for now, contact sales if you are interested in acquiring needed tarballs
+if [ -f /dunfell-cfengine-nova.tar.gz ]; then
+  cd /
+  tar xf /dunfell-cfengine-nova.tar.gz
+  cd /cfengine
+  tar xf /liblmdb.tar.gz
+fi
+
+for i in $(ls -1 /rootfs/ | grep -v -E "bin|dev|lib|lib64|proc|run|sbin|sys"); do mkdir /chroot/$i && mount -o bind /rootfs/$i /chroot/$i; done
+
+mount -o bind /cfengine /chroot/var/cfengine
+
+
+# CFENGINE_HUB_IP should be provided as an ENV var from the docker run command
+
+
+# entrypoint script for installing and running the agent to be executed from chroot jail
+cat > /chroot/entry.sh << EOF
+#!/bin/bash
+
+set -x
+
+if [ ! -f /var/cfengine/bin/cf-agent ]; then
+  sudo apt install dnsutils
+  cat /etc/resolv.conf
+  dig s3.amazonaws.com
+  cd /tmp
+  curl -O https://s3.amazonaws.com/cfengine.packages/quick-install-cfengine-enterprise.sh && bash ./quick-install-cfengine-enterprise.sh agent
+fi
+
+/var/cfengine/bin/cf-key
```

Comment:
cf-key would be done by the package, so only needed if no ppkey already
cf-key -p errors? run  cf-key